doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+
+```PfcAsymBaseTest.send```
+
+Use ```multiprocessing.Process``` instead of ```threading.Thread```.
+
+2. Looks like there is a redundancy in generating configuration file for ARP responder.
+
+```PfcAsymBaseTest.setUpArpResponder```
+
+Fix the loop to store file only once
+
+3. Need to remove dst port drop counters verification.
+
+```PfcAsymOffOnTxTest.runTest```
+
+Remove the following step which is defined by inline comment:
+```Verify that some packets are dropped on dst port, which means that Tx buffer is full```
+
+Because on some platforms packets does not count on HLL while shaper is closed and count on the HLL & SLL as soon as shaper released (but not consistently).
+
+
+## Python  modules to setup and run test
+
+### Python modules
+*New test suite will be developed:*
+
+```tests/pfc_asym/pfc_asym.py```
+
+### Global variables
+```
+PFC_GEN_FILE = "pfc_gen.py"
+PFC_FRAMES_NUMBER = 1000000
+PFC_QUEUE_INDEX = 0xff
+```
+
+### PTF test case execution
+To run existed PTF test cases there will be used ```tests/ptf_runner.py``` module
+
+### Pytest fixtures
+Preparation before test cases run will be executed in the following pytest fixtures:
+```setup, pfc_storm_template, pfc_storm_runner, deploy_pfc_gen```
+
+Description each of the fixture is below.
+
+### deploy_pfc_gen (scope="module", autouse=True)
+
+Deploy ```roles/test/files/helpers/pfc_gen.py``` to the fanout host.
+This step can be different for different platforms. Below there is description how it works for Mellanox and Arista cases, also how to add support of another platform type.
+
+**Mellanox platform**
+
+PFC packet generator is automatically deployed during fanout deployment procedure.
+
+Example of deploying fanout for Mellanox:
+```
+ansible-playbook -i lab fanout.yml -l ${FANOUT} --become --tags pfcwd_config -vvvv
+```
+
+**Arista platform**
+
+Deploy steps:
+- Ensure destination directory exists on fanout:
+"/mnt/flash/"
+- Create pfc generator file in case it doesn't exist
+"/mnt/flash/pfc_gen.py"
+- Deploy PFC generator to the fanout switch:
+Copy "roles/test/files/helpers/pfc_gen.py" to "/mnt/flash" directory
+
+**Other platforms**
+
+Tests currently support deployment of arista fanout switches, **to support other platforms:**
+
+1. Add platform specific logic to deploy pfc packet generator automatically in ```deploy_pfc_gen``` pytest fixture.
+Or manualy deploy ```roles/test/files/helpers/pfc_gen.py``` and ensure the file is available on fanout switch.
+
+2. Create ```pfc_storm_[sku].j2``` and ```pfc_storm_stop_[sku].j2``` under ```ansible/roles/test/templates/```
+to trigger pfc storm **start/stop** action.
+
+3. Set ```pfc_storm_start``` and ```pfc_storm_stop``` variables to platform-specific template names
+in ```pfc_storm_template``` pytest fixture
+
+
+### setup (scope="module")
+*Setup steps*
+
+	- Gather minigraph facts about the device
+	- Ensure topology is T0, skip tests run otherwise
```

Comment:
I believe it is better check topology at the first step, since if want to avoid any unneeded steps execution

Reply:
Fixed